### PR TITLE
Housekeeping Add net6.0 target

### PR DESCRIPTION
**Pull request by ChrisPulman**
**What kind of change does this PR introduce?**

feature

**What is the current behaviour?**

only has netstandard2.0 target

**What is the new behaviour?**

added net6.0 target for Http features

**What might this PR break?**

non expected, provides additional target

**Please check if the PR fulfils these requirements**
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Other information**:

### CODE REVIEW

File: src/Directory.build.props [hunk 1]
```diff
@@ -1,61 +1,58 @@
...
-  <ItemGroup>
-    <PackageReference Include="stylecop.analyzers" Version="1.2.0-beta.205" PrivateAssets="all" />
-    <PackageReference Include="Microsoft.CodeAnalysis.FxCopAnalyzers" Version="3.3.2" PrivateAssets="all" />
-    <PackageReference Include="Roslynator.Analyzers" Version="3.3.0" PrivateAssets="All" />
-  </ItemGroup>
-  
-  <ItemGroup>
-    <AdditionalFiles Include="$(MSBuildThisFileDirectory)stylecop.json" Link="stylecop.json" />
-  </ItemGroup>
+	<ItemGroup Condition="$(IsTestProject)">
```
Comment (glennawatson) [inline src/Directory.build.props:26]: Probably worth removing this section and moving it over to the test project directly

File: src/Directory.build.props [hunk 2]
```diff
@@ -1,61 +1,58 @@
...
+		<PackageReference Include="xunit.runner.console" Version="2.4.1" />
+		<PackageReference Include="xunit.runner.visualstudio" Version="2.4.3" />
+		<PackageReference Include="Xunit.StaFact" Version="1.0.37" />
+		<PackageReference Include="Microsoft.Reactive.Testing" Version="5.*" />
+		<PackageReference Include="PublicApiGenerator" Version="10.2.0" />
+		<PackageReference Include="coverlet.msbuild" Version="3.1.0">
+			<PrivateAssets>all</PrivateAssets>
+			<IncludeAssets>runtime; build; native; contentfiles; analyzers</IncludeAssets>
+		</PackageReference>
+		<PackageReference Include="DiffEngine" Version="8.5.2" />
```
Comment (glennawatson) [inline src/Directory.build.props:38]: Might be worth swapping over to the same approach we use on other projects now with Verify.XUnit package instead.

Comment (ChrisPulman) [reply src/Directory.build.props]: Done

File: src/Directory.build.props (hunk 1 above)
Comment (ChrisPulman) [reply src/Directory.build.props]: Moved

**Review by glennawatson — APPROVED** [review]

**Pull request merged**